File: templates/problems/16_intervals/06_min_interval_query/min_interval_query_test.go
package min_interval_query

import (
	"reflect"
	"testing"
)

func TestMinInterval(t *testing.T) {
	tests := []struct {
		name      string
		intervals [][]int
		queries   []int
		expected  []int
	}{
		{"example 1", [][]int{{1, 4}, {2, 4}, {3, 6}, {4, 4}}, []int{2, 3, 4, 5}, []int{3, 3, 1, 4}},
		{"example 2", [][]int{{2, 3}, {2, 5}, {1, 8}, {20, 25}}, []int{2, 19, 5, 22}, []int{2, -1, 4, 6}},
		{"no intervals", [][]int{}, []int{1, 2}, []int{-1, -1}},
		{"query outside all", [][]int{{1, 3}}, []int{5}, []int{-1}},
		{"single point interval", [][]int{{5, 5}}, []int{5}, []int{1}},
		{"multiple covering", [][]int{{1, 10}, {2, 5}, {3, 4}}, []int{3}, []int{2}},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := minInterval(tt.intervals, tt.queries)
			if !reflect.DeepEqual(got, tt.expected) {
				t.Errorf("minInterval(%v, %v) = %v, want %v", tt.intervals, tt.queries, got, tt.expected)
			}
		})
	}
}

File: templates/problems/16_intervals/06_min_interval_query/typescript/minIntervalQuery.test.ts
import { describe, it, expect } from 'vitest';
import { minInterval } from './minIntervalQuery';

describe('MinIntervalQuery', () => {
    it('example 1', () => {
        expect(minInterval([[1, 4], [2, 4], [3, 6], [4, 4]], [2, 3, 4, 5])).toEqual([3, 3, 1, 4]);
    });

    it('example 2', () => {
        expect(minInterval([[2, 3], [2, 5], [1, 8], [20, 25]], [2, 19, 5, 22])).toEqual([2, -1, 4, 6]);
    });

    it('no intervals', () => {
        expect(minInterval([], [1, 2])).toEqual([-1, -1]);
    });

    it('query outside all', () => {
        expect(minInterval([[1, 3]], [5])).toEqual([-1]);
    });

    it('single point interval', () => {
        expect(minInterval([[5, 5]], [5])).toEqual([1]);
    });

    it('multiple covering', () => {
        expect(minInterval([[1, 10], [2, 5], [3, 4]], [3])).toEqual([2]);
    });
});
